File: app/src/lib/models/project/ProjectDtoSimple.ts
import type { UserDtoSimple } from '../user/UserDtoSimple';

type ProjectDtoSimple = {
	id: string;
	name: string;
	likes: number;
	created: string;
	owner: UserDtoSimple;
	private: boolean;
	mainLanguage: string;
};

export type { ProjectDtoSimple };

File: server/src/main/java/com/example/projectstore/project/ProjectDtoSimple.java
package com.example.projectstore.project;

import com.example.projectstore.user.UserDtoSimple;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProjectDtoSimple {
    private UUID id;
    private String name;
    private int likes;
    private LocalDateTime created;
    private UserDtoSimple owner;
    private Boolean isPrivate;
    private String mainLanguage;
}
